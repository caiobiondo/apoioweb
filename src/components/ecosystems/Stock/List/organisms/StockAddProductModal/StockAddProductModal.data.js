import gql from 'graphql-tag';
import {
  getCycleIdFromUser,
  getCommercialStructureIdFromUser,
  getCommercialStructureTypeIdFromUser,
} from 'utils/getUserParams';

export const FetchProductQuery = gql`
  query FetchProductQuery(
    $cycleId: String!
    $productId: String!
    $commercialStructureId: Int!
    $commercialStructureTypeId: Int!
  ) {
    product(
      cycleId: $cycleId
      productId: $productId
      commercialStructureId: $commercialStructureId
      commercialStructureTypeId: $commercialStructureTypeId
    ) {
      productId
      name
      description
      price
    }
  }
`;

export const AddStockProductMutation = gql`
  mutation AddStockProduct($input: AddStockProductInput!) {
    addStockProduct(input: $input) {
      stockProduct {
        id
        productCode
        productName
        stockQuantity
      }
    }
  }
`;

export const FetchProductQueryOptions = {
  options(props) {
    console.log('FetchProductQueryOptions');
    return {
      variables: {
        productId: props.productAddSearchDebounced,
        cycleId: getCycleIdFromUser(props.user),
        commercialStructureId: getCommercialStructureIdFromUser(props.user),
        commercialStructureTypeId: getCommercialStructureTypeIdFromUser(props.user),
      },
    };
  },
};
