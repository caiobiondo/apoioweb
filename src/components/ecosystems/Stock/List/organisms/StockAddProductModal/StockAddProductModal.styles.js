import styled from 'styled-components';
import { gray200 } from 'styles/colors';
import { RobotoMedium, NaturaBold } from 'styles/typography';
import { Responsive } from '@entria/components';

export const FormWrapper = styled.div`
  flex: 1 1 55%;
`;

export const StockProductWrapper = styled.div`
  flex: 1 1 45%;
  justify-content: center;

  @media (max-width: ${Responsive.VIEWPORT.MEDIUM}px) {
    margin-bottom: 5px;
  }
`;

export const ModalTitleWrapper = styled.div`
  padding-bottom: 0 !important;

  > div {
    font-size: 25px;
  }

  @media (max-width: ${Responsive.VIEWPORT.MEDIUM}px) {
    padding-bottom: 20px !important;

    > div {
      margin-bottom: 0;
      font-size: 18px;
      line-height: 20px;
    }

    svg {
      margin-right: 10px;
    }
  }
`;

export const ModalContentWrapper = styled.div`
  display: flex;

  @media (max-width: ${Responsive.VIEWPORT.MEDIUM}px) {
    flex-direction: column-reverse;
  }
`;

export const FormInputWrapper = styled.div`
  & + & {
    margin-top: 30px;
  }

  @media (max-width: ${Responsive.VIEWPORT.MEDIUM}px) {
    & + & {
      margin-top: 10px;
    }
  }
`;

export const StockItemProductImageFallback = styled.div`
  svg {
    fill: ${gray200};
    width: 80px;
  }
`;

export const FormButtonWrapper = styled.div`
  margin-top: 54px;

  @media (max-width: ${Responsive.VIEWPORT.MEDIUM}px) {
    margin-top: 24px;

    > div {
      width: 100%;
    }
  }
`;

export const FormButtonStyles = {
  buttonStyle: {
    fontFamily: RobotoMedium,
    minWidth: '128px',
  },
  overlayStyle: {
    minWidth: '128px',
  },
  height: '50px',
};

export const titleStyle = {
  fontFamily: NaturaBold,
};

export const contentStyle = {
  maxWidth: '800px',
};

export const bodyStyle = {
  borderWidth: 0,
};
