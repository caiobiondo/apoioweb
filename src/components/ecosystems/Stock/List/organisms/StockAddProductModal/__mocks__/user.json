{
	"__type__": "net.natura.captacaoweb.dominio.Consultora",
	"codigo": 11232633,
	"nomeCompleto": "ANA PAULA CORREA NEGREIROS",
	"primeiroNome": null,
	"situacao": null,
	"dataNascimento": 299214000000,
	"documentoPessoa": [
		{
			"tipo": "2",
			"numero": "8070973873"
		}
	],
	"enderecoPessoa": [
		{
			"bairro": "CENTRO HISTORICO",
			"tipoLogradouro": "R",
			"nomeLogradouro": "DOUTOR FLORES",
			"numero": "62",
			"complemento": "5 ANDAR",
			"cep": "90020120",
			"cidade": "PORTO ALEGRE",
			"pontoReferencia": null,
			"pais": null,
			"telefoneContato": null,
			"estado": "RS"
		}
	],
	"estrutura": {
		"__type__": "net.natura.captacaoweb.dominio.Estrutura",
		"ciclo": [
			{
				"estrutura": null,
				"codigoListaPreco": null,
				"dataInicio": 1515117600000,
				"dataTermino": 1516240800000,
				"numero": 201801,
				"numeroSequencia": 306,
				"nmCicloSemAno": "01",
				"nmCicloSomenteAno": "2018",
				"labelCiclo": "01/2018",
				"labelCicloSemAno": "01"
			}
		],
		"codigo": 1035,
		"codigoTipo": 4,
		"descricaoTipo": "setor",
		"nome": null,
		"grupo": {
			"__type__": "net.natura.captacaoweb.dominio.Estrutura",
			"ciclo": null,
			"codigo": 0,
			"codigoTipo": 0,
			"descricaoTipo": null,
			"nome": null,
			"grupo": null,
			"setor": null,
			"gerenciaVenda": null,
			"regiaoEstrategica": null,
			"gerenciaMercado": null,
			"pais": null,
			"codigoENome": "0-null"
		},
		"setor": {
			"__type__": "net.natura.captacaoweb.dominio.Estrutura",
			"ciclo": null,
			"codigo": 1035,
			"codigoTipo": 4,
			"descricaoTipo": null,
			"nome": "FLORES DO SUL",
			"grupo": null,
			"setor": null,
			"gerenciaVenda": null,
			"regiaoEstrategica": null,
			"gerenciaMercado": null,
			"pais": null,
			"codigoENome": "1035-FLORES DO SUL"
		},
		"gerenciaVenda": {
			"__type__": "net.natura.captacaoweb.dominio.Estrutura",
			"ciclo": null,
			"codigo": 1,
			"codigoTipo": 3,
			"descricaoTipo": null,
			"nome": "GV RAÍZES",
			"grupo": null,
			"setor": null,
			"gerenciaVenda": null,
			"regiaoEstrategica": null,
			"gerenciaMercado": null,
			"pais": null,
			"codigoENome": "1-GV RAÍZES"
		},
		"regiaoEstrategica": {
			"__type__": "net.natura.captacaoweb.dominio.Estrutura",
			"ciclo": null,
			"codigo": 1,
			"codigoTipo": 2,
			"descricaoTipo": null,
			"nome": "SUL",
			"grupo": null,
			"setor": null,
			"gerenciaVenda": null,
			"regiaoEstrategica": null,
			"gerenciaMercado": null,
			"pais": null,
			"codigoENome": "1-SUL"
		},
		"gerenciaMercado": {
			"__type__": "net.natura.captacaoweb.dominio.Estrutura",
			"ciclo": null,
			"codigo": 1,
			"codigoTipo": 1,
			"descricaoTipo": null,
			"nome": "GER. DE MERCADO SUL",
			"grupo": null,
			"setor": null,
			"gerenciaVenda": null,
			"regiaoEstrategica": null,
			"gerenciaMercado": null,
			"pais": null,
			"codigoENome": "1-GER. DE MERCADO SUL"
		},
		"pais": {
			"__type__": "net.natura.captacaoweb.dominio.Estrutura",
			"ciclo": null,
			"codigo": 1,
			"codigoTipo": 0,
			"descricaoTipo": null,
			"nome": "Brasil",
			"grupo": null,
			"setor": null,
			"gerenciaVenda": null,
			"regiaoEstrategica": null,
			"gerenciaMercado": null,
			"pais": null,
			"codigoENome": "1-Brasil"
		},
		"codigoENome": "1035-null"
	},
	"papel": [
		{
			"codigo": 5,
			"nomeTipo": "PROMOTORA DE VENDAS ",
			"dataIncio": 965876400000,
			"dataFim": null,
			"cicloInicio": 200011,
			"cicloFim": 0,
			"codigoMotivo": 0
		},
		{
			"codigo": 12,
			"nomeTipo": "COLABORADOR",
			"dataIncio": 965271600000,
			"dataFim": null,
			"cicloInicio": 0,
			"cicloFim": 0,
			"codigoMotivo": 0
		}
	],
	"bloqueio": null,
	"idSituacaoBloqueio": null,
	"idMotivoBloqueio": null,
	"cdCanalCaptacao": 8,
	"listaNiveisConsultora": null,
	"pedidoIncluido": null,
	"cdCanalCaptacaoReal": 8,
	"quantidadePontos": null,
	"identificadorKit": 0,
	"numeroPedidoKit": 19262545,
	"dataInicioRelacionamento": null,
	"permissaoCan": false,
	"cdPapelAtivo": 5,
	"dcPapelAtivo": "PROMOTORA DE VENDAS ",
	"ciclo": null,
	"cnStatus": null,
	"pedido": null,
	"credito": {
		"vlPontuacaoTotal": null,
		"vlPontuacaoMinImpersonacao": null,
		"existeVlPontuacaoMinImpersonacao": false,
		"nmPromocaoImpersonacao": null,
		"dsPromocaoImpersonacao": null,
		"vlCreditoDisponivelAtual": null,
		"promocao": null,
		"promocaoImpersonacao": null,
		"vlCreditoExcedenteAtual": null,
		"vlCreditoPgtoAVistaAtual": null,
		"situacaoBloqueio": false,
		"situacaoPentende": false,
		"vlPontuacaoMin": 0,
		"vlCreditoTotal": 220,
		"pcCreditoAdicional": 125.0,
		"vlCreditoAdicional": null,
		"vlCreditoDisponivel": null,
		"pcCreditoExcedente": 125.0,
		"vlCreditoExcedente": null,
		"idOrigemPromocao": null,
		"nmPromocao": null,
		"dsPromocao": null,
		"vlCreditoTotalAtual": null,
		"vlCreditoDisponivelCalc": null,
		"vlCreditoExcedenteCalc": null,
		"exibiCreditoExcedente": false,
		"vlCreditoUtilizado": null,
		"pontuacaoMinimaPromocionada": false,
		"vlPontosKitParcial": 0,
		"vlPontuacaoComprometido": null,
		"situacaoPagtoAVista": null,
		"vlCreditoDisponivelIncr": 0,
		"calculaPontosKitParcial": 0,
		"pontuacaoMaxima": 220
	},
	"bloqueioCAN": false,
	"dataPrevisaoEntrega": null,
	"proximaDataPrevisaoEntrega": null,
	"centro": null,
	"cdZoneamentoEntrega": "RS0149005",
	"formasPagamento": null,
	"email": "MARCELO.VERRI.BORGES@EVERIS.COM",
	"numeroTelefone": "982047307",
	"dddTelefone": "11",
	"ddiTelefone": null,
	"formaPagamento": null,
	"formaEnvioBoleto": {
		"codigo": null,
		"descricao": null
	},
	"listaFormasEnvioBoleto": null,
	"codigoRiscoConsultora": 0,
	"msgRiscoConsultoraFraudulenta": null,
	"alertaCan": false,
	"listaTreinamentosBloqueio": null,
	"listaTreinamentosAlerta": null,
	"condicaoComercial": null,
	"listaCondicaoComercial": null,
	"perfis": null,
	"possuiPendenciaPorDebito": false,
	"codigoSegmentoCanal": 2,
	"codigoSegmentoCanalConsulta": null,
	"idTipoCn": 1,
	"dadosAPI": true,
	"bloquearListaKit": false,
	"pontosPeriodo": 0,
	"cdCanalResgateValePontos": "CANAL_RESGATE_TODOS",
	"qtValePontosExcedentesPromocao": 1,
	"qtValePontosExcedentesTratativa": 1,
	"produtoPromocaoSubstituta": null,
	"consultoraIndicadas": null,
	"comGarantiaCompraPendente": false,
	"liberadoCotas": false,
	"liberadoPtMinima": false,
	"candidataAprovada": false,
	"cnTeste": false,
	"dtPedidoCNTeste": null,
	"cdPapelReal": 5,
	"dcPapelReal": "PROMOTORA DE VENDAS ",
	"captacaoRetroativa": false,
	"dataCaptacaoRetroativa": null,
	"bloqueioEstrategia": false,
	"estrategia": {
		"dataAberturaCan": null,
		"dataAberturaWeb": null,
		"dataAberturaWebGr": null,
		"cdPapelAtivo": 5,
		"listaBloqueiosEstrategia": [],
		"listaLiberacoesEstrategia": null,
		"setDataAberturaWebGr": null,
		"liberadoCan": false,
		"liberadoWeb": false,
		"liberadoWebGr": false,
		"bloqueioCan": false,
		"bloqueioWeb": false,
		"bloqueioWebGr": false
	},
	"parametroCicloRetroativo": null,
	"parametroRetroatividade": null,
	"parametroSaldao": [],
	"parametroSobMedida": [],
	"desaplicarPromoCancelarPedido": false,
	"promoCancelarPedidoDesaplicada": false,
	"situacaoPentende": false,
	"situacaoPgtoAVista": false,
	"permissaoPgtoAVista": false,
	"dataRetornoAnalise": null,
	"selecionadoCicloAtual": false,
	"utilizarCachePromocao": false,
	"colaboradora": false,
	"dataImpersonacao": 1516311277279,
	"kitInicioConsultora": null,
	"acao": null,
	"listaKitInicio": null,
	"creditoInexistente": false,
	"ajudaOnlineVideo": false,
	"idade": null,
	"quantidadeTempoCasa": null,
	"pontuacaoMinimaCnIniciante": false,
	"listaMaterialNaoEmpenhaEstoque": null,
	"listaProdutosVendaNaoPromocionada": null,
	"listaProdutosTrocaDePara": null,
	"descricaoRiscoConsultora": null,
	"cdFuncaoNoPedido": null,
	"descricaoFuncaoNoPedido": null,
	"meioPagamento": null,
	"dataUltimaCompra": 1507431600000,
	"codEstruturaCNTeste": null,
	"codTipoEstruturaCNTeste": null,
	"numCicloCNTeste": null,
	"codCanalCaptacaoCNTeste": null,
	"aceitaPreVenda": false,
	"diaIdealCaptacao": null,
	"diaIdealProximaCaptacao": null,
	"possuiPedidoPendentePorDebito": false,
	"pessoaJuridica": false,
	"segmentacaoCorCN": null,
	"codigoSegmentoKitInicio": null,
	"cdSistemaOrigem": 5,
	"faixasLucratividade": null,
	"tratarCNOComoLider": false,
	"listaKitDesejo": null,
	"cnFacilitadora": false,
	"listaPromocoesRealTime": [],
	"aplicarPromoRealTime": true,
	"cnteste": false,
	"previsaoDtEntregaCalculada": false,
	"cartaoHabilitadoCNInicio": false,
	"boletoHabilitadoCNInicio": false
}
