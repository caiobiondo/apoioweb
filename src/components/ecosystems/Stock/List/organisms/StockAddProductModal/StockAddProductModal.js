import React, { Component } from 'react';
import { Modal, FormInput, FormButton, FlatButton, Dialog } from 'natura-ui';
import PropTypes from 'prop-types';
import { dialogContainer, dialogContent, dialogTitle, dialogActions } from 'styles/dialog';
import { translate } from 'locale';
import {
  bodyStyle,
  contentStyle,
  FormButtonStyles,
  FormButtonWrapper,
  FormWrapper,
  FormInputWrapper,
  titleStyle,
  ModalTitleWrapper,
  ModalContentWrapper,
  StockProductWrapper,
} from './StockAddProductModal.styles';
import { fetchProduct, AddStockProductMutation } from './StockAddProductModal.data';
import { graphql } from 'react-apollo';
import { StockProductsQuery } from '../ListTable/ListTable.data';
import debounce from 'lodash.debounce';
import {
  getCycleIdFromUser,
  getCommercialStructureIdFromUser,
  getCommercialStructureTypeIdFromUser,
} from 'utils/getUserParams';
import StockProduct from '../../molecules/StockProduct';
import SectionTitle from 'components/molecules/SectionTitle';

export class StockAddProductModal extends Component {
  state = {
    successOpened: false,
    productQty: 1,
    productCode: '',
    loadedProduct: null,
    loadingProduct: false,
    lastProductCode: '',
    importing: false,
  };

  loadProduct = debounce(productCode => {
    if (!productCode) {
      this.setState({ loadedProduct: null });
      return;
    }

    this.setState({
      loadingProduct: true,
      lastProductCode: productCode,
    });

    fetchProduct(productCode, this.props.user)
      .catch(() => {
        return null;
      })
      .then(product => {
        if (productCode === this.state.lastProductCode) {
          this.setState({
            loadingProduct: false,
            loadedProduct: product,
          });
        }
      });
  }, 1000);

  onChangeProductAddSearch = (event, productCode) => {
    this.setState({ productCode });
    this.loadProduct(productCode);
  };

  onSubmited = () => {
    this.setState({ successOpened: true, importing: false });
    this.props.handleClose();
  };

  onClose = () => {
    this.setState({ successOpened: false });
  };

  onChangeProductQty = (event, productQty) => {
    this.setState({ productQty });
  };

  onSubmit = () => {
    const product = this.state.loadedProduct;
    const imageUrl = `http://rede.natura.net/image/sku/145x145/${product.productId}_1.jpg`;
    this.setState({ importing: true });
    this.props
      .mutate({
        variables: {
          input: {
            productCode: product.productId,
            stockQuantity: this.state.productQty,
            productName: product.name,
            productDescription: product.description,
            productImage: imageUrl,
            productPrice: product.price,
            cycleId: getCycleIdFromUser(this.props.user),
            commercialStructureId: getCommercialStructureIdFromUser(this.props.user),
            commercialStructureTypeId: getCommercialStructureTypeIdFromUser(this.props.user),
          },
        },
        refetchQueries: [
          {
            query: StockProductsQuery,
            variables: {
              limit: 10,
              offset: 0,
              filter: '',
            },
          },
        ],
      })
      .then(this.onSubmited);
  };

  renderSuccessDialog = () => {
    const title = translate('stockProductSuccessfullyAdded');
    const actions = [
      <FlatButton
        label={translate('ok')}
        primary={true}
        onClick={this.onClose}
        labelStyle={dialogActions}
      />,
    ];

    return (
      <Dialog
        key="successDialog"
        title={title}
        actions={actions}
        modal={false}
        open={this.state.successOpened}
        onRequestClose={this.onClose}
        contentStyle={dialogContainer}
        bodyStyle={dialogContent}
        titleStyle={dialogTitle}
      />
    );
  };

  renderProduct = () => {
    return (
      <StockProductWrapper>
        <StockProduct product={this.state.loadedProduct} loading={this.state.loadingProduct} />
      </StockProductWrapper>
    );
  };

  allowSubmit = () => {
    return (
      this.state.loadedProduct &&
      !this.state.loadingProduct &&
      this.state.productQty &&
      this.state.productQty !== '0' &&
      !this.state.importing
    );
  };

  renderForm = () => {
    return (
      <FormWrapper>
        <FormInputWrapper>
          <FormInput
            onChange={this.onChangeProductAddSearch}
            name="productCode"
            label={translate('stockProductCodeLabel')}
            value={this.state.productCode}
          />
        </FormInputWrapper>
        <FormInputWrapper>
          <FormInput
            onChange={this.onChangeProductQty}
            name="qty"
            label={translate('stockProductQuantityLabel')}
            value={this.state.productQty}
          />
        </FormInputWrapper>
        <FormButtonWrapper>
          <FormButton
            primary
            disabled={!this.allowSubmit()}
            label={this.renderAddButtonLabel()}
            onClick={this.onSubmit}
            {...FormButtonStyles}
          />
        </FormButtonWrapper>
      </FormWrapper>
    );
  };

  renderAddButtonLabel = () => {
    if (this.state.importing) {
      return translate('stockProductAdding');
    }

    return translate('stockProductAdd');
  };

  renderTitle() {
    return (
      <ModalTitleWrapper>
        <SectionTitle value="stockProductAddModalTitle" iconName="ico_flask" />
      </ModalTitleWrapper>
    );
  }

  render() {
    return [
      <Modal
        open={this.props.opened}
        showCloseButton={true}
        onCloseClick={this.props.handleClose}
        title={this.renderTitle()}
        titleStyle={titleStyle}
        contentStyle={contentStyle}
        bodyStyle={bodyStyle}
        autoScrollBodyContent={true}
      >
        <ModalContentWrapper>
          {this.renderForm()}
          {this.renderProduct()}
        </ModalContentWrapper>
      </Modal>,
      this.renderSuccessDialog(),
    ];
  }
}

StockAddProductModal.propTypes = {
  handleClose: PropTypes.func,
  opened: PropTypes.bool,
  user: PropTypes.object,
};

export default graphql(AddStockProductMutation)(StockAddProductModal);
