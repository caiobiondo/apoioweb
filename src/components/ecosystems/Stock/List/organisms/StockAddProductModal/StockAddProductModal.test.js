import React from 'react';
import { StockAddProductModal } from './StockAddProductModal';
import { shallow } from 'enzyme';
import user from './__mocks__/user.json';

jest.mock('lodash.debounce', () => jest.fn(fn => fn));

const setup = propOverrides => {
  const props = Object.assign(
    {
      user: user,
      opened: false,
      handleClose: jest.fn(),
      mutate: jest.fn().mockReturnValue(Promise.resolve()),
      client: {
        query: jest.fn().mockReturnValue(
          Promise.resolve({
            data: {
              product: {
                description: 'new product',
                name: 'new product',
                price: '10',
                productId: '1',
              },
            },
          }),
        ),
      },
    },
    propOverrides,
  );

  const result = shallow(<StockAddProductModal {...props} />);

  return {
    props,
    result,
  };
};

describe('StockAddProductModal', () => {
  it('renders correctly', () => {
    // given
    const props = {
      opened: true,
    };

    // when
    const { result } = setup(props);

    // then
    expect(result).toMatchSnapshot();
  });

  describe('when searching for a product', () => {
    it("loads a product by it's code", () => {
      // given
      const newProps = {
        opened: true,
      };
      const code = '1';

      // when
      const { props, result } = setup(newProps);
      result.instance().setState({ lastProductCode: code });
      result.instance().loadProduct(code);

      // then
      expect(props.client.query).toBeCalled();
    });

    describe('when there is no code', () => {
      it('does not load a product', () => {
        // given
        const newProps = {
          opened: true,
        };
        const code = null;

        // when
        const { props, result } = setup(newProps);
        result.instance().loadProduct(code);

        // then
        expect(props.client.query).not.toBeCalled();
      });
    });
  });

  describe('when submitting', () => {
    describe('on success', () => {
      it('adds the selected product to stock', () => {
        // given
        const product = {
          description: 'new product',
          name: 'new product',
          price: '10',
          productId: '1',
        };
        const newProps = {
          opened: true,
        };

        // when
        const { props, result } = setup(newProps);
        result.instance().setState({ loadedProduct: product });
        result.instance().onSubmit();

        // then
        expect(props.mutate).toBeCalled();
      });
    });

    describe('on failure', () => {
      it('opens feedback dialog', async () => {
        // given
        const product = {
          description: 'new product',
          name: 'new product',
          price: '10',
          productId: '1',
        };
        const newProps = {
          opened: true,
          mutate: jest.fn().mockReturnValue(Promise.reject()),
        };

        // when
        const { props, result } = setup(newProps);
        result.instance().setState({ loadedProduct: product });
        await result.instance().onSubmit();

        // then
        expect(props.mutate).toBeCalled();
        expect(result.instance().state.openFeedbackDialog).toBeTruthy();
      });
    });
  });
});
