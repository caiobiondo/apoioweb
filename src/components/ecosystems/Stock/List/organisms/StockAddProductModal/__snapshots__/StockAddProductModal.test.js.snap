// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StockAddProductModal renders correctly 1`] = `
Array [
  <Modal
    autoScrollBodyContent={true}
    bodyStyle={
        Object {
            "borderWidth": 0,
          }
    }
    contentStyle={
        Object {
            "maxWidth": "800px",
          }
    }
    key="stockAddProductModal"
    onCloseClick={[Function]}
    open={true}
    showCloseButton={true}
    title={
        <StockAddProductModalstyles__ModalTitleWrapper>
            <SectionTitle
                iconName="ico_flask"
                value="stockProductAddModalTitle"
            />
        </StockAddProductModalstyles__ModalTitleWrapper>
    }
    titleStyle={
        Object {
            "fontFamily": "NaturaSans-Bold",
          }
    }
/>,
  <Dialog
    actions={
        Array [
            <FlatButton
              disabled={false}
              fullWidth={false}
              label="OK"
              labelPosition="after"
              labelStyle={
                    Object {
                          "fontFamily": "Roboto",
                        }
              }
              onClick={[Function]}
              onKeyboardFocus={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onTouchStart={[Function]}
              primary={true}
              secondary={false}
        />,
          ]
    }
    autoDetectWindowHeight={true}
    autoScrollBodyContent={false}
    bodyStyle={
        Object {
            "fontFamily": "Roboto",
            "padding": "12px 25px",
          }
    }
    contentStyle={
        Object {
            "maxWidth": "450px",
          }
    }
    key="successDialog"
    modal={false}
    onRequestClose={[Function]}
    open={false}
    repositionOnUpdate={true}
    titleStyle={
        Object {
            "fontWeight": 800,
          }
    }
/>,
]
`;
